pilot/pkg/bootstrap/server.go

```diff
@@ -1281,3 +1287,60 @@ func (s *Server) initStatusManager(_ *PilotArgs) {
 		return nil
 	})
 }
+
+// initLocalSecretListener creates a new secret listener to listen for
+// secrets which represent new multi cluster participating clusters in a mesh.
+func (s *Server) initLocalSecretListener(args *PilotArgs) {
+	// Check if the istiod's kube client API endpoint is the same as the local cluster API endpoint
+	// If they are the same, then the Istiod is in cluster Istiod, otherwise, it will be an
+	// external istiod running outside of the config cluster, the localKubeClient and the
+	// istiod kube client endpoint should have been different.
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Errorf("Could not get istiod incluster configuration: %v", err)
+		return
+	}
+	log.Info("Successfully retrieved incluster config.")
+
+	localKubeClient, err := kubelib.NewClient(kubelib.NewClientConfigForRestConfig(config))
+	if err != nil {
+		log.Errorf("Could not create a client to access local cluster API server: %v", err)
+		return
+	}
+	log.Infof("Successfully created in cluster kubeclient at %s", localKubeClient.RESTConfig().Host)
+	if s.kubeClient.RESTConfig().Host == localKubeClient.RESTConfig().Host {
+		log.Info("The config cluster and local cluster are the same")
+		return
+	}
+
+	// Now create secret controller to listen for the secret in the istiod running
+	// namespace
+	namespace := env.RegisterStringVar("POD_NAMESPACE", "", "").Get()
+	if namespace == "" {
+		log.Error("Could not get istiod namespace.")
+		return
+	}
+	log.Infof("Successfully retrieved Istiod namespace: %s", namespace)
+
+	// create a new secret controller using the same store that the other secret controller
+	// also uses.
+	controller := multicluster.NewControllerWithClusterStore(localKubeClient, namespace,
```

Comment:
I donot think it is good to introduce another branch to init multi cluster secret controller